pkg/grafana/dashboard-handler.go

```diff
@@ -372,36 +375,56 @@ func (h *DashboardHandler) DashboardJSONGetHandler(p grizzly.Server) http.Handle
 
 func (h *DashboardHandler) DashboardJSONPostHandler(p grizzly.Server) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-
-		dash := map[string]interface{}{}
+		resp := struct {
+			Dashboard map[string]any `json:"dashboard"`
+		}{}
 		content, _ := io.ReadAll(r.Body)
-		err := json.Unmarshal(content, &dash)
+		err := json.Unmarshal(content, &resp)
 		if err != nil {
-			http.Error(w, "Error parsing JSON", 400)
+			grizzly.SendError(w, "Error parsing JSON", err, 400)
 			return
 		}
-
-		resource, err := grizzly.NewResource(h.APIVersion(), h.Kind(), "dummy", dash)
+		resource, err := grizzly.NewResource(h.APIVersion(), h.Kind(), "dummy", resp.Dashboard)
 		if err != nil {
-			http.Error(w, "Error creating resource", 400)
+			grizzly.SendError(w, "Error creating resource", err, 400)
 			return
 		}
-		uid, err := h.GetUID(resource)
+		uid, err := h.GetSpecUID(resource)
 		if err != nil {
-			http.Error(w, "Error getting dashboard UID", 400)
+			grizzly.SendError(w, "Error getting dashboard UID", err, 400)
+			return
+		}
+		if uid == "" {
+			grizzly.SendError(w, "Dashboard has no UID", fmt.Errorf("dashboard has no UID"), 400)
 			return
 		}
 		resource.SetMetadata("name", uid)
+		resource.SetSpecString("uid", uid)
 
 		out, _, _, err := grizzly.Format(p.Registry, p.ResourcePath, &resource, p.OutputFormat, p.OnlySpec)
 		if err != nil {
-			http.Error(w, "Error formatting content", 400)
+			grizzly.SendError(w, "Error formatting content", err, 400)
+			return
+		}
+
+		resources, err := p.Parser.Parse()
+		if err != nil {
+			grizzly.SendError(w, "Error parsing existing resources", err, 400)
```

Comment:
```suggestion
			grizzly.SendError(w, "Error parsing existing resources", err, 500)
```

Same here. Though ideally we wouldn't need to parse everything here (this should be changed... eventually)